src/puppetlabs/trapperkeeper/services/status/ringutils.clj

```diff
@@ -1,48 +1,85 @@
 ;; TODO: Much of this code is copied from other projects. It should
 ;; probably be moved into a shared library or some such.
 (ns puppetlabs.trapperkeeper.services.status.ringutils
-  (:require [clojure.tools.logging :as log]
+  (:require [schema.core :as schema]
+            [clojure.tools.logging :as log]
+            [cheshire.core :as json]
+            [ring.util.response :as response]
             [slingshot.slingshot :refer [try+]]
             [puppetlabs.kitchensink.core :as ks]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ResponseType (schema/enum :json :plain))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
+(defmacro with-json-response [status & body]
+  `(-> (do ~@body)
+       json/encode
+       response/response
+       (response/status ~status)
+       (response/content-type "application/json; charset=utf-8")))
+
+(defmacro with-plain-response [status & body]
+  `(-> (do ~@body)
+       response/response
+       (response/status ~status)
+       (response/content-type "text/plain; charset=utf-8")))
```

Comment:
they're from another library where they actually wrap bodies. I didn't go back and make them functions. I can do that.